R/derive_param_bor.R

```diff
@@ -0,0 +1,541 @@
+#'
+#'   *Required or Optional:* Required
+#'
+#' @examples
+#'
+#' library(magrittr)
+#'
+#' # Create ADSL dataset
+#' adsl <- tibble::tribble(
+#'   ~USUBJID, ~TRTSDTC,
+#'   "1",      "2020-01-01",
+#'   "2",      "2019-12-12",
+#'   "3",      "2019-11-11",
+#'   "4",      "2019-12-30",
+#'   "5",      "2020-01-01",
+#'   "6",      "2020-02-02",
+#'   "7",      "2020-02-02",
+#'   "8",      "2020-04-01"
+#'   ) %>%
+#'   dplyr::mutate(TRTSDT = lubridate::ymd(TRTSDTC),
+#'                 STUDYID = "XX1234")
+#'
+#' # Create ADRS dataset
+#' ovr_obs <- tibble::tribble(
+#'   ~USUBJID, ~ADTC,        ~AVALC,
+#'   "1",      "2020-01-01", "PR",
+#'   "1",      "2020-02-01", "CR",
+#'   "1",      "2020-02-16", "NE",
+#'   "1",      "2020-03-01", "CR",
+#'   "1",      "2020-04-01", "SD",
+#'   "2",      "2020-01-01", "SD",
+#'   "2",      "2020-02-01", "PR",
+#'   "2",      "2020-03-01", "SD",
+#'   "2",      "2020-03-13", "CR",
+#'   "3",      "2019-11-12", "CR",
+#'   "3",      "2019-12-02", "CR",
+#'   "3",      "2020-01-01", "SD",
+#'   "4",      "2020-01-01", "PR",
+#'   "4",      "2020-03-01", "SD",
+#'   "4",      "2020-04-01", "SD",
+#'   "4",      "2020-05-01", "PR",
+#'   "4",      "2020-05-15", "NON-CR/NON-PD",
+#'   "5",      "2020-01-01", "PR",
+#'   "5",      "2020-01-10", "SD",
+#'   "5",      "2020-01-20", "PR",
+#'   "5",      "2020-05-15", "NON-CR/NON-PD",
+#'   "6",      "2020-02-06", "PR",
+#'   "6",      "2020-02-16", "CR",
+#'   "6",      "2020-03-30", "PR",
+#'   "6",      "2020-04-12", "PD",
+#'   "6",      "2020-05-01", "CR",
+#'   "6",      "2020-06-01", "CR",
+#'   "7",      "2020-02-06", "PR",
+#'   "7",      "2020-02-16", "CR",
+#'   "7",      "2020-04-01", "NE"
+#'   ) %>%
+#'   dplyr::mutate(PARAMCD = "OVR")
+#'
+#' pd_obs <-
+#'   dplyr::bind_rows(tibble::tribble(
+#'     ~USUBJID, ~ADTC,        ~AVALC,
+#'     "2",      "2020-03-01", "Y",
+#'     "4",      "2020-02-01", "Y"
+#'     ) %>%
+#'     dplyr::mutate(PARAMCD = "PD"))
+#'
+#' adrs <- dplyr::bind_rows(ovr_obs, pd_obs) %>%
+#'   dplyr::mutate(ADT     = lubridate::ymd(ADTC),
+#'                 STUDYID = "XX1234") %>%
+#'   dplyr::select(-ADTC) %>%
+#'   admiral::derive_vars_merged(
+#'     dataset_add = adsl,
+#'     by_vars     = dplyr::vars(STUDYID, USUBJID),
+#'     new_vars    = dplyr::vars(TRTSDT) )
+#'
+#' pd_date <- admiral::date_source(
+#'   dataset_name = "adrs",
+#'   date         = ADT,
+#'   filter       = PARAMCD == "PD")
+#'
+#' aval_fun_pass <- function(arg) {
+#'         dplyr::case_when(
+#'           arg == "CR" ~ 11,
+#'           arg == "PR" ~ 22,
+#'           arg == "SD" ~ 33,
+#'           arg == "NON-CR/NON-PD" ~ 44,
+#'           arg == "PD" ~ 55,
+#'           arg == "NE" ~ 66,
+#'           arg == "MISSING" ~ 77,
+#'           TRUE ~ NA_real_
+#'         ) }
+#'
+#' # Derive best overall response parameter
+#' derive_param_bor(
+#'   adrs,
+#'   dataset_adsl     = adsl,
+#'   filter_source    = PARAMCD == "OVR",
```

Comment:
please also add `& ANL01FL == "Y"` here - i want users to get used to seeing this convention given its such an important part of our workflow, but the ADRS vignette will explain more anyway

Reply:
Updated

Reply:
@rossfarrugia just pushed

Reply:
thanks @sgorm123 these are all resolved in the latest commit i just saw